### PR TITLE
ogr2ogr: make -select tokenize only on comma (and not on space), and honour double-quoting 

### LINKED ISSUE #9613: -select cannot deal with blanks, even if quoted
### What is the bug?

man ogr2ogr says
>       -select <field_list>
              Comma-delimited  list  of fields from input layer to copy to the
              new layer. A field is skipped if  mentioned  previously  in  the
              list  even  if  the  input layer has duplicate field names. (De-
              faults to all; any field is skipped if a subsequent  field  with
              same  name  is  found.) Geometry fields can also be specified in
              the list.

Alas, currently `-select` cannot deal with fields with blanks in their names, even if quoted.

### Versions and provenance

GDAL 3.8.4, released 2024/02/08

### PR DESCRIPTION
**Pull request by rouault**
(fixes #9613)

### CODE REVIEW

File: doc/source/programs/ogr2ogr.rst [hunk 1]
```diff
@@ -104,11 +104,28 @@ output coordinate system or even reprojecting the features during translation.
...
-    Comma-delimited list of fields from input layer to copy to the new layer. A
-    field is skipped if mentioned previously in the list even if the input
-    layer has duplicate field names. (Defaults to ``all``; any field is skipped
-    if a subsequent field with same name is found.) Geometry fields can also be
-    specified in the list.
+    Comma-delimited list of fields from input layer to copy to the new layer.
+
+    Starting with GDAL 3.9, field names with spaces, commas or double-quote
+    should be surrounded with a starting and ending double-quote character, and
+    double-quote characters in a field name should be escaped with antislash.
```
Comment (dbaston) [inline doc/source/programs/ogr2ogr.rst:111]:
```suggestion
    double-quote characters in a field name should be escaped with backslash.
```

File: autotest/utilities/test_ogr2ogr_lib.py [hunk 2]
```diff
@@ -195,6 +195,37 @@ def test_ogr2ogr_lib_sel_fields_empty():
     assert lyr.GetLayerDefn().GetFieldCount() == 0
 
 
+###############################################################################
+# Test selFields with field names with spaces
```
Comment (dbaston) [inline autotest/utilities/test_ogr2ogr_lib.py:199]:
```suggestion
# Test selectFields with field names with spaces
```

File: doc/source/programs/ogr2ogr.rst (hunk 1 above)
Comment (rouault) [reply doc/source/programs/ogr2ogr.rst]: Frenchism... fixed.

File: autotest/utilities/test_ogr2ogr_lib.py (hunk 2 above)
Comment (rouault) [reply autotest/utilities/test_ogr2ogr_lib.py]: fixed

**Pull request merged**